Source/Scene/ClassificationModel.js

```diff
@@ -0,0 +1,1869 @@
+    }
+
+    LoadResources.prototype.getBuffer = function(bufferView) {
+        return getSubarray(this.buffers[bufferView.buffer], bufferView.byteOffset, bufferView.byteLength);
+    };
+
+    LoadResources.prototype.finishedBuffersCreation = function() {
+        return ((this.pendingBufferLoads === 0) &&
+                (this.vertexBuffersToCreate.length === 0) &&
+                (this.indexBuffersToCreate.length === 0));
+    };
+
+    LoadResources.prototype.finishedProgramCreation = function() {
+        return ((this.pendingShaderLoads === 0) && (this.programsToCreate.length === 0));
+    };
+
+    LoadResources.prototype.finishedEverythingButTextureCreation = function() {
+        var finishedPendingLoads =
+            (this.pendingBufferLoads === 0) &&
+            (this.pendingShaderLoads === 0);
+        var finishedResourceCreation =
+            (this.vertexBuffersToCreate.length === 0) &&
+            (this.indexBuffersToCreate.length === 0) &&
+            (this.programsToCreate.length === 0);
+
+        return finishedPendingLoads && finishedResourceCreation;
+    };
+
+    LoadResources.prototype.finished = function() {
+        return this.finishedEverythingButTextureCreation();
+    };
+
+    ///////////////////////////////////////////////////////////////////////////
+
+    /**
+     * A 3D model for classifying other 3D assets based on glTF, the runtime asset format for WebGL, OpenGL ES, and OpenGL.
+     *
+     * @alias ClassificationModel
+     * @constructor
+     *
+     * @private
+     *
+     * @param {Object} [options] Object with the following properties:
+     * @param {Object|ArrayBuffer|Uint8Array} options.gltf The object for the glTF JSON or an arraybuffer of Binary glTF defined by the KHR_binary_glTF extension.
+     * @param {String} [options.basePath=''] The base path that paths in the glTF JSON are relative to.
+     * @param {Boolean} [options.show=true] Determines if the model primitive will be shown.
+     * @param {Matrix4} [options.modelMatrix=Matrix4.IDENTITY] The 4x4 transformation matrix that transforms the model from model to world coordinates.
+     * @param {Boolean} [options.debugShowBoundingVolume=false] For debugging only. Draws the bounding sphere for each draw command in the model.
+     * @param {Boolean} [options.debugWireframe=false] For debugging only. Draws the model in wireframe.
+     * @param {DistanceDisplayCondition} [options.distanceDisplayCondition] The condition specifying at what distance from the camera that this model will be displayed.
+     * @param {ClassificationType} [options.classificationType] What this model will classify.
+     *
+     * @exception {DeveloperError} bgltf is not a valid Binary glTF file.
+     * @exception {DeveloperError} Only glTF Binary version 1 is supported.
+     *
+     * @demo {@link http://cesiumjs.org/Cesium/Apps/Sandcastle/index.html?src=3D%20Models.html|Cesium Sandcastle Models Demo}
+     */
+    function ClassificationModel(options) {
```

Comment:
This still needs tests, right?  We should test this as an isolated unit, and not just rely on the 3D Tiles tests.

Reply:
I started writing tests here but it required duplicating a ton of code from `Batched3DModel3DTileContent` in the spec.

Reply:
Is it possible that when you create a common file for both `Model` implementations that it will because easier to write unit tests since you can also write tests against that?